core/src/main/java/com/graphhopper/routing/weighting/custom/BooleanToValueEntry.java

```diff
@@ -20,18 +20,60 @@
 import com.graphhopper.routing.profiles.BooleanEncodedValue;
 import com.graphhopper.util.EdgeIteratorState;
 
+import java.util.Map;
+
+import static com.graphhopper.routing.weighting.custom.CustomWeighting.CATCH_ALL;
+import static com.graphhopper.routing.weighting.custom.EnumToValueEntry.getReturnValue;
+
 final class BooleanToValueEntry implements EdgeToValueEntry {
     private final BooleanEncodedValue bev;
     private final double value, elseValue;
 
-    public BooleanToValueEntry(BooleanEncodedValue bev, double value, double elseValue) {
+    private BooleanToValueEntry(BooleanEncodedValue bev, double value, double elseValue) {
         this.bev = bev;
         this.value = value;
         this.elseValue = elseValue;
     }
 
+    /**
+     * Example map:
+     * <pre>
+     * get_off_bike:
+     *   true: 0.4
+     *   false: 0.9 // optional and default is 1, equivalent to "*": 0.9
+     * </pre>
+     */
+    static EdgeToValueEntry create(String name, BooleanEncodedValue encodedValue, Map<Object, Object> map,
+                                   double defaultValue, double minValue, double maxValue) {
+        if (map.isEmpty())
+            throw new IllegalArgumentException("Empty map for " + name);
+
+        if (map.containsKey(CATCH_ALL) && (map.containsKey("false") || map.containsKey(false)))
```

Comment:
should we really allow the key to be of type boolean. hm well that is what it is , but all the other keys are strings so this would be a bit more consistent?

Reply:
I'm not sure and tbh I have not even tested this (I'll do!) but the big problem could be that people do not quote the `false` like they do not need to quote the `motorway`.

Reply:
Ah thats true I forgot you do not have to explicitly quote string parameters without special chars like `motorway` either

Reply:
Turns out jacks forces the keys to be string even if you specify it like just `false` in the yaml. So this is indeed not necessary.

Reply:
Ok good then :) Btw can we then delete this branch? Github allows removing the branches (so they do not show up anymore, but its even possible to restore them). Maybe there is even a default setting for github to delete branches after they are merged from a PR?

Reply:
Done :) !

> Maybe there is even a default setting for github to delete branches after they are merged from a PR?

Did not find something like this yet

Reply:
Ah, or is it this:

> After pull requests are merged, you can have head branches deleted automatically.
> Automatically delete head branches
> Deleted branches will still be able to be restored.

Reply:
Have enabled it for now :)